### ISSUE TITLE
Pull in uv_getrusage

### ISSUE THREAD

**Issue by wraithan** [numbat-metrics/numbat-process#6]
[`uv_getrusage()`](http://docs.libuv.org/en/v1.x/misc.html#c.uv_getrusage) provides lots of information about the process from the kernel's point of view. It provides the [`uv_rusage_t`](http://docs.libuv.org/en/v1.x/misc.html#c.uv_rusage_t) struct of data. 

Haven't looked for a native module that exposes this yet (it would be trivial to build so whether it exists or not, lets assume it does). There is a [proposal](https://github.com/nodejs/node/pull/1568) to add this to node itself, but it is waiting on additional [Windows support](https://github.com/libuv/libuv/issues/342) before landing in core.

I know you are getting some of the data from the `procfs` module but this gathers it all in one place and some (though not all) of the values work in windows as well. Windows support isn't important to me for a server monitoring product but it may be to some people. Additionally it has some good stuff you aren't yet collecting like context switches, break down of memory consumption, and page stats.

Happy to either pull in an established module or write one. If this is more invasive than you'd like then feel free to close wontfix and I can build it as a separate gatherer.
